faiss_vector_posting.go

```diff
@@ -387,13 +387,80 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, requiresFiltering bool
 						vectorIDsToInclude = append(vectorIDsToInclude, docVecIDMap[uint32(id)]...)
 					}
 
-					scores, ids, err := vecIndex.SearchWithIDs(qVector, k,
-						vectorIDsToInclude, params)
+					// Gives the vector IDs for each cluster.
+					clusterAssignment, _ := vecIndex.GetClusterAssignment()
+					// Accounting for flat index
+					if len(clusterAssignment) == 0 {
+						scores, ids, err := vecIndex.SearchWithIDs(qVector, k,
+							vectorIDsToInclude, params)
+						if err != nil {
+							return nil, err
+						}
+
+						addIDsToPostingsList(rv, ids, scores)
+						return rv, nil
+					}
+
+					// Converting to roaring bitmap for ease of intersect ops with
+					// the set of eligible doc IDs.
+					centroidVecIDMap := make(map[int64]*roaring.Bitmap)
+					for centroidID, vecIDs := range clusterAssignment {
+						if _, exists := centroidVecIDMap[centroidID]; !exists {
+							centroidVecIDMap[centroidID] = roaring.NewBitmap()
+						}
+						for _, vecID := range vecIDs {
+							centroidVecIDMap[centroidID].Add(uint32(vecID))
+						}
+					}
+
+					eligibleVecIDsBitmap := roaring.NewBitmap()
+					for _, eligibleDocID := range eligibleDocIDs {
+						vecIDs := docVecIDMap[uint32(eligibleDocID)]
```

Comment:
Edge case to consider:
not all eligible docs have a vector assigned to them - confirm if these are filtered out by the AND.

Reply:
If this is still an open task for you, let's follow up separately.

Reply:
nope, just left it open to add unit testing - which i'm doing separately.